### PR TITLE
PLAT-2547: [mf-execution] Research performance issues on table

### PR DESCRIPTION
**Pull request by chavlji**
Adding new prop to popover that will enable us to wait with popover render until popover is rendered.

[JIRA ticket](https://reciprocitylabs.atlassian.net/browse/PLAT-2527)

### CODE REVIEW

File: src/components/zen-dropdown/zen-dropdown.tsx [hunk 1]
```diff
@@ -268,8 +268,9 @@ export class ZenDropdown {
     this.opened = this.popover.visible;
   }
 
-  onBeforeToggle(): void {
+  onBeforeToggle(): Promise<void> {
     this.matchDropdownWidth();
+    return new Promise(resolve => resolve());
   }
```
Comment (homer0) [inline src/components/zen-dropdown/zen-dropdown.tsx:274]:
```suggestion
  async onBeforeToggle(): Promise<void> {
    this.matchDropdownWidth();
  }
```

1. For an already resolved promise, you could use `return Promise.resolve();`
2. But if you add `async`, the method/function gets automatically wrapped in a promise, and all promises return a new promise, so there's no need to manually return one.

**Review by homer0 — COMMENTED** [review]
Sorry, but I don't understand how a new callback prop would be the solution, why the listener was removed, and how the implementation would look like.

We can probably discuss this during tomorrow offline.

**Comment by chavlji** [comment]
@homer0 I'll explain it on a daily.

**Review by homer0 — APPROVED** [review]
I would say, fix the tests and go for it :P.

**Pull request merged**